modules/opensearch/includes/opensearch.search.inc

```diff
@@ -109,19 +109,29 @@ function opensearch_search_search(TingSearchRequest $ting_query) {
     $query_parts[] = $free_text_query;
   }
 
-  // Add material filter.
-  if (!empty($ting_query->getMaterialFilter())) {
-    $material_filter_string = implode(' OR ', $ting_query->getMaterialFilter());
-    $query_parts[] .= $material_filter_string;
-  }
-
   // Add field filter.
   if (!empty($ting_query->getFieldFilters())) {
     $render = new StatementGroupRender(new DingProviderStrategy());
     $field_filters = $render->renderStatements($ting_query->getFieldFilters());
     $query_parts[] = $field_filters;
   }
 
+  // Add material filter, this has to be the very last thing we do as we have
+  // to handle the filter differently query_parts is empty.
```

Comment:
I really appreciate the comment but I do not see how it is implemented. I would have expected a check whether `$query_parts` is `empty()` or not.

Either the comment is no longer relevant and should be removed or the subsequent code needs to be updated.